src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -77,11 +77,31 @@ public void visitToken(DetailAST ast) {
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
 
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!containCaseBeforeOrAfterDefault(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
+                        || defaultGroupAST.getColumnNo()
+                            != lastGroupAST.getColumnNo())) {
                 log(ast, MSG_KEY);
             }
         }
     }
+
+    /**
+     * It returns true if after or before default there is a case or it
+     * returns false if before or after default there is no case statement.
```

Comment:
Javadoc description sounds too redundant. `A is B, not A is not B`.
Please make javadoc more meaningful like `Examines ast to see if it is surrounded by a 'case' on either side.`